app/uk/gov/hmrc/hec/services/scheduleService/HecTaxCheckExtractionService.scala

```diff
@@ -129,50 +125,50 @@ class HecTaxCheckExtractionServiceImpl @Inject() (
   private def toFormattedString(i: Int) = f"$i%04d"
 
   private def createHecFile(
-    hecTaxCheckList: HECTaxCheckFileBodyList,
-    count: Int,
+    limit: Int,
```

Comment:
think it'd probably be better to keep the db limit separate from `maxTaxChecksPerFile` config so they can be tweaked independently if required

Reply:
We are keeping that limit because of the maxTaxChecksPerFile, so that whatever number is required for one file, we can process it in one go. if we keep the, separate then we will have two loops, one for the DB pagination , other for the record process

Reply:
ah yes, ok, that makes sense

separate issue, but there's a [default limit on the batch size in mongo](https://docs.mongodb.com/manual/reference/limits/#std-label-limit-bson-document-size) (16MB) so mongo by default does [paging](https://docs.mongodb.com/manual/tutorial/iterate-a-cursor/), so if `maxTaxChecksPerFile` docs together exceeds this limit, we'd start to see some missed tax checks - I guess we just need to be conscious of this when we are updating `maxTaxChecksPerFile`. 16MB is quite a large amount of data, so unlikely we'll hit the limit

Reply:
yeah, it's unlikely to hit that number